src/nupic/regions/PyRegion.py

```diff
@@ -177,6 +177,15 @@ def getOutputElementCount(self, name):
     name: the name of the output
     """
 
+  def getAlgorithm(self):
```

Comment:
I don't think this makes sense to add. The return value for a region of unknown type is an unknown object. It would only make sense if we required all regions to have an "algorithm" object that subclasses the some common base class.